php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+        BITWISE_AND_EXPR(),
+        b.zeroOrMore(f.newTuple67(
+          b.token(PHPPunctuator.XOR),
+          BITWISE_AND_EXPR()))));
+  }
+
+  public ExpressionTree BITWISE_OR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.BITWISE_OR).is(
+      f.bitwiseOrExpr(
+        BITWISE_XOR_EXPR(),
+        b.zeroOrMore(f.newTuple68(
+          b.token(PHPPunctuator.OR),
+          BITWISE_XOR_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.CONDITIONAL_AND).is(
+      f.logicalAndExpr(
+        BITWISE_OR_EXPR(),
+        b.zeroOrMore(f.newTuple69(
+          b.firstOf(b.token(PHPPunctuator.ANDAND), b.token(PHPKeyword.AND)),
+          BITWISE_OR_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_XOR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.ALTERNATIVE_CONDITIONAL_XOR).is(
+      f.logicalXorExpr(
+        CONDITIONAL_AND_EXPR(),
+        b.zeroOrMore(f.newTuple70(
+          b.token(PHPKeyword.XOR),
+          CONDITIONAL_AND_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_OR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.CONDITIONAL_OR).is(
+      f.logicalOrExpr(
+        CONDITIONAL_XOR_EXPR(),
+        b.zeroOrMore(f.newTuple71(
+          b.firstOf(b.token(PHPPunctuator.OROR), b.token(PHPKeyword.OR)),
+          CONDITIONAL_XOR_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.CONDITIONAL_EXPR).is(
+      f.completeConditionalExpr(
+        CONDITIONAL_OR_EXPR(),
+        b.optional(
+          f.newConditionalExpr(b.token(QUERY), b.optional(CONDITIONAL_EXPR()), b.token(COLON), CONDITIONAL_EXPR())
+        )));
+  }
+
+  public ExpressionTree EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EXPRESSION).is(
+      CONDITIONAL_EXPR());
+  }
+
+  public ExpressionTree COMMON_SCALAR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.COMMON_SCALAR).is(
+      b.firstOf(
+        f.heredocLiteral(b.token(PHPLexicalGrammar.HEREDOC)),
+        NUMERIC_LITERAL(),
+        STRING_LITERAL(),
+        f.booleanLiteral(b.token(PHPLexicalGrammar.BOOLEAN_LITERAL)),
+        f.nullLiteral(b.token(PHPLexicalGrammar.NULL)),
+        f.magicConstantLiteral(b.firstOf(
+          b.token(PHPLexicalGrammar.CLASS_CONSTANT),
+          b.token(PHPLexicalGrammar.FILE_CONSTANT),
+          b.token(PHPLexicalGrammar.DIR_CONSTANT),
+          b.token(PHPLexicalGrammar.FUNCTION_CONSTANT),
+          b.token(PHPLexicalGrammar.LINE_CONSTANT),
+          b.token(PHPLexicalGrammar.METHOD_CONSTANT),
+          b.token(PHPLexicalGrammar.NAMESPACE_CONSTANT),
+          b.token(PHPLexicalGrammar.TRAIT_CONSTANT)))));
+  }
+
+  public LiteralTree NUMERIC_LITERAL() {
+    return b.<LiteralTree>nonterminal(Kind.NUMERIC_LITERAL).is(
+      f.numericLiteral(b.token(PHPLexicalGrammar.NUMERIC_LITERAL)));
+  }
+
+  public ExpressionTree STRING_LITERAL() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)